tools/get_all_unlabeled_resources.sh

```diff
@@ -0,0 +1,249 @@
+    "deployments"
+    "replicasets"
+    "statefulsets"
+    "daemonsets"
+    "configmaps"
+    # "secrets"
+    "serviceaccounts"
+    "persistentvolumeclaims"
+    "jobs"
+    "cronjobs"
+    "ingresses"
+    "networkpolicies"
+    "validatingwebhookconfigurations"
+    "mutatingwebhookconfigurations"
+)
+
+UNLABELED_RESOURCES=()
+
+# Function to check if a name matches any keyword pattern
+matches_keyword() {
+    local name=$1
+    local keywords=$2
+    local name_lower=$(echo "${name}" | tr '[:upper:]' '[:lower:]')
+    
+    # Split keywords by | (compatible with both bash and zsh)
+    IFS='|' read -ra keyword_array <<< "${keywords}"
+    for keyword in "${keyword_array[@]}"; do
+        if echo "${name_lower}" | grep -q "${keyword}"; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+# Function to check if a label is valid for a resource
+# Handles special cases where alternative labels are acceptable
+is_label_valid() {
+    local name=$1
+    local label_value=$2
+    local expected_label=$3
+    local name_lower=$(echo "${name}" | tr '[:upper:]' '[:lower:]')
+    
+    # Check if it matches the expected label
+    if [[ "${label_value}" == "${expected_label}" ]]; then
+        return 0
+    fi
+    
+    # Special case 1: 'observability-cluster-metrics' resources can have 'microsoft-iotoperations-observability-cluster-metrics' label
+    if echo "${name_lower}" | grep -q "observability-cluster-metrics"; then
+        if [[ "${label_value}" == "microsoft-iotoperations-observability-cluster-metrics" ]]; then
+            return 0
+        fi
+    fi
+    
+    # Special case 2: 'akri-adr' resources can still have 'microsoft-iotoperations-akri' label
+    if echo "${name_lower}" | grep -q "akri-adr"; then
+        if [[ "${label_value}" == "microsoft-iotoperations-akri" ]]; then
+            return 0
+        fi
+    fi
+    
+    # Special case 3: 'opc-ua-broker' or 'opc-opcuabroker' resources can have 'microsoft-iotoperations-opcuabroker' label
+    if echo "${name_lower}" | grep -qE "opc-ua-broker|opc-opcuabroker"; then
+        if [[ "${label_value}" == "microsoft-iotoperations-opcuabroker" ]]; then
+            return 0
+        fi
+    fi
+    
+    return 1
+}
+
+# Check each keyword-label pair
+# Compatible iteration over associative array keys (works in both bash and zsh)
+for keywords_pattern in "${!KEYWORD_LABEL_MAP[@]}"; do
+    expected_label="${KEYWORD_LABEL_MAP[$keywords_pattern]}"
+    
+    echo "========================================"
+    echo -e "Checking resources with keywords: ${BOLD}${keywords_pattern}${RESET}"
+    echo -e "Expected label: ${BOLD}app.kubernetes.io/name=${expected_label}${RESET}"
+    echo "========================================"
+    echo ""
+    
+    for resource in "${RESOURCE_TYPES[@]}"; do
+        echo "Checking ${resource}..."
+        
+        # Get all resources
+        all_resources=$(kubectl get ${resource} --all-namespaces -o wide 2>/dev/null)
```

Comment:
nit, hopefully the output order of `-o wide` doesn't change. `-o json` + `jq` may be more reliable.